### ISSUE TITLE
False positive on rule "Azurerm Container Registry Must Contain Associated Locks"

### ISSUE THREAD

**Issue by LvffY** [Checkmarx/kics#4578]
### Expected Behavior

My playbook should not fall into the error **Azurerm Container Registry Must Contain Associated Locks**

### Actual Behavior

KICS analysis return (what I think to be) a false positive on the rule **Azurerm Container Registry Must Contain Associated Locks**

### Steps to Reproduce the Problem

1.  Have the following playbook

```yaml
---
- name: Deploy shared resources
  hosts: infra
  environment:
    AZURE_CLIENT_ID: "{{ azure.client_id }}"
    AZURE_SECRET: "{{ azure.secret }}"
    AZURE_SUBSCRIPTION_ID: "{{ azure.subscription_id }}"
    AZURE_TENANT: "{{ azure.tenant }}"

  tasks:

    - name: "Create shared resource group {{ global.shared.rg_name }}"
      azure.azcollection.azure_rm_resourcegroup:
        name: "{{ global.shared.rg_name }}"
        location: "{{ global.location }}"
        state: present

    - name: "Create ACR {{ global.shared.acr_name }}"
      azure.azcollection.azure_rm_containerregistry:
        name: "{{ global.shared.acr_name }}"
        resource_group: "{{ global.shared.rg_name }}"
        state: present
      register: acr

    - name: "Create lock for ACR {{ global.shared.acr_name }}"
      azure.azcollection.azure_rm_lock:
        managed_resource_id: "{{ acr.id }}"
        name: "{{ global.shared.acr_name }}_lock"
        level: can_not_delete
```
2.  In `checkmarx/kics:debian` run the analysis `/app/bin/kics scan -p group_vars/backends/vars.yml -o ${PWD} --report-formats json,sarif --ignore-on-exit results --verbose --log-level=DEBUG`

***IMHO:*** *This is due to the fact of both Ansible module works. You can't use the exact same name in the module <b>azure_rm_containerregistry</b> and in <b>azure_rm_lock</b>*

### Specifications

-   Version: 1.4.8
-   Platform: Debian
-   Subsystem: Ansible

Please let me know if something is not clear

**Comment by rafaela-soares** [comment]
Hello, @LvffY 🙂 Thank you so much for reaching us and reporting the bug. 

We will add the `managed_resource_id` verification. The query was only checking for 'resource_group'.
